chain-addr/src/lib.rs

```diff
@@ -77,6 +77,15 @@ impl KindType {
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct Address(pub Discrimination, pub Kind);
```

Comment:
You've already implemented the expliclit fields, perhaps, and add an `Address::new` constructor?